lifecycle-operator/apis/lifecycle/v1beta1/keptntaskdefinition_types.go

```diff
@@ -0,0 +1,198 @@
+}
+
+type RuntimeSpec struct {
+	// FunctionReference allows to reference another KeptnTaskDefinition which contains the source code of the
+	// function to be executes for KeptnTasks based on this KeptnTaskDefinition. This can be useful when you have
+	// multiple KeptnTaskDefinitions that should execute the same logic, but each with different parameters.
+	// +optional
+	FunctionReference FunctionReference `json:"functionRef,omitempty"`
+	// Inline allows to specify the code that should be executed directly in the KeptnTaskDefinition, as a multi-line
+	// string.
+	// +optional
+	Inline Inline `json:"inline,omitempty"`
+	// HttpReference allows to point to an HTTP URL containing the code of the function.
+	// +optional
+	HttpReference HttpReference `json:"httpRef,omitempty"`
+	// ConfigMapReference allows to reference a ConfigMap containing the code of the function.
+	// When referencing a ConfigMap, the code of the function must be available as a value of the 'code' key
+	// of the referenced ConfigMap.
+	// +optional
+	ConfigMapReference ConfigMapReference `json:"configMapRef,omitempty"`
+	// Parameters contains parameters that will be passed to the job that executes the task as env variables.
+	// +optional
+	Parameters TaskParameters `json:"parameters,omitempty"`
+	// SecureParameters contains secure parameters that will be passed to the job that executes the task.
+	// These will be stored and accessed as secrets in the cluster.
+	// +optional
+	SecureParameters SecureParameters `json:"secureParameters,omitempty"`
+	// CmdParameters contains parameters that will be passed to the command
+	// +optional
+	CmdParameters string `json:"cmdParameters,omitempty"`
+}
+
+type ConfigMapReference struct {
+	// Name is the name of the referenced ConfigMap.
+	// +optional
+	Name string `json:"name,omitempty"`
+}
+
+type FunctionReference struct {
+	// Name is the name of the referenced KeptnTaskDefinition.
+	// +optional
+	Name string `json:"name,omitempty"`
+}
+
+type Inline struct {
+	// Code contains the code of the function.
+	// +optional
+	Code string `json:"code,omitempty"`
+}
+
+type HttpReference struct {
+	// Url is the URL containing the code of the function.
+	// +optional
+	Url string `json:"url,omitempty"`
+}
+
+type ContainerSpec struct {
+	*v1.Container `json:",inline"`
+}
+
+type AutomountServiceAccountTokenSpec struct {
+	Type *bool `json:"type"`
+}
+type ServiceAccountSpec struct {
+	Name string `json:"name"`
+}
+
+// KeptnTaskDefinitionStatus defines the observed state of KeptnTaskDefinition
+type KeptnTaskDefinitionStatus struct {
+	// Function contains status information of the function definition for the task.
+	// +optional
+	Function FunctionStatus `json:"function,omitempty"`
+}
+
+type FunctionStatus struct {
+	// ConfigMap indicates the ConfigMap in which the function code is stored.
+	// +optional
+	ConfigMap string `json:"configMap,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
```

Comment:
please move the `storageversion` to the task definition in `v1alpha3` - the storage versions will change in a follow up ticket when we adapt the controller to use v1beta1

Reply:
the same also applies for the other types in `v1beta1` - they should not include the `// +kubebuilder:storageversion` marker yet